File: Baekjoon/문제풀이/후위 표기식2/yuri.java

import java.util.*;

public class yuri {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();
        String postfix = sc.nextLine();
        double[] values = new double[n];
        for (int i = 0; i < n; i++) values[i] = sc.nextDouble(); // 각 변수의 값을 입력받음

        Stack<Double> stack = new Stack<>();
        for (char ch : postfix.toCharArray()) {
            if (Character.isLetter(ch)) {
                stack.push(values[ch - 'A']); // 변수 값을 스택에 푸시
            } else {
                double b = stack.pop(), a = stack.pop();
                switch (ch) {
                    case '+': stack.push(a + b); break; // 덧셈
                    case '-': stack.push(a - b); break; // 뺄셈
                    case '*': stack.push(a * b); break; // 곱셈
                    case '/': stack.push(a / b); break; // 나눗셈
                }
            }
        }

        System.out.printf("%.2f", stack.pop()); // 결과 출력
    }
}

File: Baekjoon/문제풀이/후위 표기식2/yuri.py
n = int(input()) # 피연산자 개수
postfix = input() # 후위 표기식
values = [int(input()) for _ in range(n)]  # 변수 값 입력, 정수 변환

stack = []
for ch in postfix:
    if ch.isalpha():
        stack.append(values[ord(ch) - ord('A')])  # 아스키 코드 값으로 변환해서 빼기 / A=65 
                                                  # A부터 순서대로 변수 선언하기 때문에 A값을 빼면 변수가 입력된 리스트에서의 인덱스 값을 구할 수 있음
    else:
        b = stack.pop()
        a = stack.pop()
        if ch == '+':
            stack.append(a + b)  # 덧셈
        elif ch == '-':
            stack.append(a - b)  # 뺄셈
        elif ch == '*':
            stack.append(a * b)  # 곱셈
        elif ch == '/':
            stack.append(a / b)  # 나눗셈

print(f"{stack[0]:.2f}")  # 결과 출력
